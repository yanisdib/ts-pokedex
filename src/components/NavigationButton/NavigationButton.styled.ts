import styled from 'styled-components';


export const Button = styled.span`
    display: flex;
    justify-content: center;
    align-items: center;
    min-width: 45px;
    min-height: 45px;
    border: 2px solid red;
    border-radius: 50px;
    cursor: pointer;
    
    svg {
        height: 25px;
        fill: red;
        stroke: red;
        stroke-width: 2;
    }
`;
