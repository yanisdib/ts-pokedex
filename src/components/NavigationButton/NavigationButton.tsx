import { ChevronLeft, ChevronRight } from '@styled-icons/heroicons-outline';

import { Button } from './NavigationButton.styled';


interface Props {
    isNext?: boolean;
    onClick?: () => void;
}


function NavigationButton({
    isNext = true,
    onClick
}: Props): JSX.Element {
    return (
        <Button onClick={onClick}>
            {isNext ? <ChevronRight /> : <ChevronLeft />}
        </Button>
    );
}


export default NavigationButton;