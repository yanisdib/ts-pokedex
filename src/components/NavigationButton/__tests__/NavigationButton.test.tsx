import { render } from '@testing-library/react';

import NavigationButton from '../NavigationButton';


describe('NavigationButton component', () => {
    test('should render without crashing', () => {
        const { container } = render(<NavigationButton />);

        expect(container).toMatchSnapshot();
    });
    
    // test('should render ChevronRight by default', () => {
    //     const { getByTestId } = render(<NavigationButton />);

    //     expect(getByTestId('test-right')).toBeInTheDocument();
    // });

    // test('should render ChevronRight if button is next', () => {
    //     const { getByTestId } = render(<NavigationButton isNext={true} />);

    //     expect(getByTestId('test-right')).toBeInTheDocument();
    // });

    // test('should render ChevronLeft if button is false', () => {
    //     const { getByTestId } = render(<NavigationButton isNext={false} />);

    //     expect(getByTestId('test-left')).toBeInTheDocument();
    // });
});