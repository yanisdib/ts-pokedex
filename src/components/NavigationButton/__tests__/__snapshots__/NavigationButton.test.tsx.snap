// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationButton component should render without crashing 1`] = `
<div>
  <span
    class="sc-bczRLJ jXCQsR"
  >
    <svg
      aria-hidden="true"
      class="StyledIconBase-ea9ulj-0 lbJwfL"
      fill="none"
      focusable="false"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m9 5 7 7-7 7"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </span>
</div>
`;
